src/iceberg/avro/avro_writer.cc

```diff
@@ -73,9 +76,22 @@ class AvroWriter::Impl {
     return {};
   }
 
-  Status Write(ArrowArray /*data*/) {
-    // TODO(xiao.dong) convert data and write to avro
-    // total_bytes_+= written_bytes;
+  Status Write(ArrowArray data) {
+    ArrowSchema arrow_schema;
+    ICEBERG_RETURN_UNEXPECTED(ToArrowSchema(*write_schema_, &arrow_schema));
```

Comment:
Consider caching `ArrowSchema` in the `WriterContext` to avoid recreating it every time. This is an expensive operation.